### PR TITLE
feat: MakesFederatedCall servant combinator

### PR DESCRIPTION
**Pull request by isovector**
## Checklist

 - [ ] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [ ] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: services/brig/src/Brig/API/Internal.hs [hunk 1]
```diff
@@ -175,10 +175,10 @@ userAPI =
     :<|> deleteLocale
     :<|> getDefaultUserLocale
 
-authAPI :: (Member GalleyProvider r, CallsFed 'Brig "on-user-deleted-connections") => ServerT BrigIRoutes.AuthAPI (Handler r)
+authAPI :: (Member GalleyProvider r) => ServerT BrigIRoutes.AuthAPI (Handler r)
```
Comment (isovector) [inline services/brig/src/Brig/API/Internal.hs:178]: This PR lets us safely eliminate this `CallsFed` constraint.

File: libs/wire-api/src/Wire/API/MakesFederatedCall.hs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  ShowComponent 'Cargohold = "cargohold"
+
+-- | 'MakesFederatedCall' annotates the swagger documentation with an extension
+-- tag @x-wire-makes-federated-calls-to@.
+instance (HasSwagger api, KnownSymbol name, KnownSymbol (ShowComponent comp)) => HasSwagger (MakesFederatedCall comp name :> api :: *) where
+  toSwagger _ =
+    toSwagger (Proxy @api)
+      & applyTags
+        [ Tag
+            "x-wire-makes-federated-call-to"
```
Comment (fisx) [inline libs/wire-api/src/Wire/API/MakesFederatedCall.hs]: i just built this branch and tried this out, but came up empty:

```
$ curl -q http://localhost:8082/api/swagger.json | jq . | grep x-wire-makes-federated-call-to
```

am i holding it wrong?

Comment (isovector) [reply libs/wire-api/src/Wire/API/MakesFederatedCall.hs]: I'm not sure how to run the server, but dumping the output of `Wire.API.Routes.Internal.Brig.swaggerDoc` gives me the following output:

[swagger.json.txt](https://github.com/isovector/wire-server/files/10280571/swagger.json.txt)

which gives the following output

```
$ cat /tmp/swagger.json.txt| jq . | grep x-wire-makes-federated-call-to
          "x-wire-makes-federated-call-to"
          "x-wire-makes-federated-call-to"
      "name": "x-wire-makes-federated-call-to"
```

But upon viewing this in a swagger viewer, it looks like tags might be the wrong place for this information:

2022-12-21-112522_585x281_scrot

**Comment by isovector** [comment]
Resultant swagger for brig after this change:

[swagger.json.txt](https://github.com/isovector/wire-server/files/10281617/swagger.json.txt)

**Comment by fisx** [comment]
could you push this branch to wireapp/wire-server and make a PR from there (either rewire this one or create a new one)?  then we can have the CI take a look at things.

it would be interesting to understand where the output you're working with goes, and what goes into `http://localhost:8082/api/swagger.json` (which i think is the one used [here](https://staging-nginz-https.zinfra.io/api/swagger-ui), but i'm ont sure).

File: nix/haskell-pins.nix [hunk 3]
```diff
@@ -172,6 +172,13 @@ let
         sha256 = "1w23yz2iiayniymk7k4g8gww7268187cayw0c8m3bz2hbnvbyfbc";
       };
     };
+    swagger2 = {
```
Comment (supersven) [inline nix/haskell-pins.nix:175]: Why do we need exactly this version of `swagger2`?

Comment (mdimjasevic) [inline nix/haskell-pins.nix:175]: Why this Swagger2 pin? I suppose it provides something that currently doesn't exist in the version that is used.

Comment (isovector) [reply nix/haskell-pins.nix:175]: @mdimjasevic asked the same thing. The swagger specification supports [extensions](https://swagger.io/docs/specification/2-0/swagger-extensions/), but `swagger2` doesn't. Julia and I decided to patch swagger2 with the functionality and have [upstreamed it](https://github.com/GetShopTV/swagger2/pull/245), but it's unclear if the library is maintained.

**Comment by isovector** [comment]
@fisx 
> could you push this branch to wireapp/wire-server and make a PR from there (either rewire this one or create a new one)? then we can have the CI take a look at things.

will do!

> it would be interesting to understand where the output you're working with goes, and what goes into `http://localhost:8082/api/swagger.json` (which i think is the one used [here](https://staging-nginz-https.zinfra.io/api/swagger-ui), but i'm ont sure).

After investigating, this PR currently only changes an internal brig endpoint's definition, and the internal API's swagger isn't served anywhere.

**Comment by isovector** [comment]
Reopened at https://github.com/wireapp/wire-server/pull/2950

**Pull request closed without merging**